### PR TITLE
Issue #297: Separate password confirm from strength indicator. Add show/hide password.

### PR DESCRIPTION
**Pull request by quicksketch**
Fixes https://github.com/backdrop/backdrop-issues/issues/297.

Replaces https://github.com/backdrop/backdrop/pull/735.

### CODE REVIEW

**Comment by herbdool** [comment]
I don't see any test site. Is it still working?

**Comment by serundeputy** [comment]
@quicksketch When the `show password` toggle is checked I receive a js error:

```js
user.js:76 Uncaught TypeError: Cannot read property 'preventDefault' of undefined
    at passwordToggle (user.js:76)
    at HTMLFormElement.<anonymous> (user.js:88)
    at HTMLFormElement.dispatch (jquery.js?v=1.12.4:3)
    at HTMLFormElement.r.handle (jquery.js?v=1.12.4:3)
```

I've added a check here https://github.com/quicksketch/backdrop/pull/5/files#diff-4ad1c261e20bfda2a28f08d04fd746deR76 to prevent this error.

Otherwise works as advertised!

File: core/modules/user/js/user.js [hunk 1]
```diff
@@ -1,45 +1,114 @@
...
+          $passwordWrapper.removeClass('password-hidden').addClass('password-shown');
+          $passwordToggle.text(passwordToggleSettings.toggleHideTitle);
+        }
+        else {
+          // Set the element to password and set the toggle to be "Show".
+          $passwordInput.attr('type', 'password');
+          $passwordWrapper.removeClass('password-shown').addClass('password-hidden');
+          $passwordToggle.text(passwordToggleSettings.toggleShowTitle);
+        }
+        e.preventDefault();
```
Comment (serundeputy) [inline core/modules/user/js/user.js:76]: add check here: https://github.com/quicksketch/backdrop/pull/5/files#diff-4ad1c261e20bfda2a28f08d04fd746deR76

**Review by serundeputy — DISMISSED** [review]
Add check to `e.preventDefaul()` to prevent js errors.

**Comment by serundeputy** [comment]
closing to rebuild sandy sandbox. ⏳

**Comment by serundeputy** [comment]
looks good works good thanks @quicksketch !

**Pull request merged**